spirv_cross.cpp

```diff
@@ -5465,6 +5465,52 @@ void Compiler::analyze_interlocked_resource_usage()
 	}
 }
 
+// Helper function
+bool Compiler::check_internal_recursion(SPIRType &type, std::unordered_set<uint32_t> &checked_ids)
+{
+	// If this is not the base type (eg. it's a pointer or array), tunnel down
+	if (type.parent_type)
+		return check_internal_recursion(get<SPIRType>(type.parent_type), checked_ids);
+
+	if ( !(type.basetype == SPIRType::Struct) )
+		return false;
+
+	if (has_extended_decoration(type.self, SPIRVCrossDecorationNonRecursiveStruct))
```

Comment:
Do we *really* need two more decorations here? It bloats memory usage in the meta structs. These decorations are only used in local analysis, can just keep this information in a map while emitting entry point.

Reply:
Good point. When I put it together, I wasn't sure how often it might be called, and I looked for a short-circuit to always recursing. 

I've added a `set` to hold the info now, and populate it with one round of recursion tests.